### PR TITLE
Feature: Added support for multiple tag search

### PR DESCRIPTION
**Pull request by nilayjoshi89**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Closes #issue #9671
- Related #issue #9671

**Details of Changes**
- Fixed - FolderSearch to look up for exact tags

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [x] Tested functionality manually

### CODE REVIEW

File: src/Files.Uwp/Filesystem/Search/FolderSearch.cs [hunk 1]
```diff
@@ -191,13 +191,16 @@ private async Task AddItemsAsyncForLibrary(LibraryLocationItem library, IList<Li
         private async Task SearchTagsAsync(string folder, IList<ListedItem> results, CancellationToken token)
         {
             //var sampler = new IntervalSampler(500);
-            var tagName = AQSQuery.Substring("tag:".Length);
-            var tags = FileTagsSettingsService.SearchTagsByName(tagName);
-            if (!tags.Any())
+            var tags = AQSQuery.Substring("tag:".Length)?.Split(",")?.Where(t => !string.IsNullOrWhiteSpace(t))
+                .SelectMany(t => FileTagsSettingsService.GetTagsByName(t));
```
Comment (hez2010) [inline src/Files.Uwp/Filesystem/Search/FolderSearch.cs:195]:
```suggestion
            var tags = AQSQuery.Substring("tag:".Length)?.Split(",").Where(t => !string.IsNullOrWhiteSpace(t))
                .SelectMany(t => FileTagsSettingsService.GetTagsByName(t), (_, t) => t.Uid).ToHashSet();
```

File: src/Files.Uwp/Filesystem/Search/FolderSearch.cs [hunk 2]
```diff
@@ -191,13 +191,16 @@ private async Task AddItemsAsyncForLibrary(LibraryLocationItem library, IList<Li
...
+            if (tags?.Any() != true)
             {
                 return;
             }
-            var matches = FileTagsHelper.DbInstance.GetAllUnderPath(folder).Where(x => tags.Select(t => t.Uid).Intersect(x.Tags).Any());
+
+            var matches = FileTagsHelper.DbInstance.GetAllUnderPath(folder)
+                            .Where(x => tags.All(t => x.Tags.Contains(t.Uid)));
```
Comment (hez2010) [inline src/Files.Uwp/Filesystem/Search/FolderSearch.cs:202]:
```suggestion
                            .Where(x => tags.All(x.Tags.Contains));
```

**Review by hez2010 — APPROVED** [review]
You can use `HashSet` to avoid unnecessary computations. 
Otherwise LGTM.

**Review by yaichenbaum — APPROVED** [review]
Thank you for your contribution!

**Pull request merged**